packages/happychat-connection/src/use-happychat-auth.ts

```diff
@@ -4,49 +4,47 @@ import { useQuery } from 'react-query';
 import wpcomRequest from 'wpcom-proxy-request';
 import type { HappychatSession, HappychatUser, User, HappychatAuth } from './types';
 
-export default function useHappychatAuth( enabled = true ) {
-	return useQuery< HappychatAuth, typeof Error >(
-		'getHappychatAuth',
-		async () => {
-			const user: User = await wpcomRequest( {
-				path: '/me',
-				apiVersion: '1.1',
-			} );
+export async function requestHappyChatAuth() {
+	const user: User = await wpcomRequest( {
+		path: '/me',
+		apiVersion: '1.1',
+	} );
 
-			const url: string = config( 'happychat_url' );
-			const locale = getLocaleSlug();
+	const url: string = config( 'happychat_url' );
+	const locale = getLocaleSlug();
 
-			const happychatUser: HappychatUser = {
-				signer_user_id: user.ID,
-				locale,
-				groups: [ 'WP.com' ],
-				skills: {
-					product: [ 'WP.com' ],
-				},
-			};
-			const session: HappychatSession = await wpcomRequest( {
-				path: '/happychat/session',
-				apiVersion: '1.1',
-				method: 'POST',
-			} );
-			const { session_id, geo_location } = session;
-			happychatUser.geoLocation = geo_location;
+	const happychatUser: HappychatUser = {
+		signer_user_id: user.ID,
+		locale,
+		groups: [ 'WP.com' ],
+		skills: {
+			product: [ 'WP.com' ],
+		},
+	};
+	const session: HappychatSession = await wpcomRequest( {
+		path: '/happychat/session',
+		apiVersion: '1.1',
+		method: 'POST',
+	} );
+	const { session_id, geo_location } = session;
+	happychatUser.geoLocation = geo_location;
 
-			const sign: { jwt: string } = await wpcomRequest( {
-				path: '/jwt/sign',
-				apiVersion: '1.1',
-				method: 'POST',
-				body: { payload: JSON.stringify( { user, session_id } ) },
-			} );
+	const sign: { jwt: string } = await wpcomRequest( {
+		path: '/jwt/sign',
+		apiVersion: '1.1',
+		method: 'POST',
+		body: { payload: JSON.stringify( { user, session_id } ) },
+	} );
 
-			const { jwt } = sign;
+	const { jwt } = sign;
 
-			return { url, user: { jwt, ...happychatUser } };
-		},
-		{
-			refetchOnWindowFocus: false,
-			keepPreviousData: true,
-			enabled,
-		}
-	);
+	return { url, user: { jwt, ...happychatUser }, fullUser: user };
+}
+
+export default function useHappychatAuth( enabled = true ) {
+	return useQuery< HappychatAuth, typeof Error >( 'getHappychatAuth', requestHappyChatAuth, {
+		refetchOnWindowFocus: false,
+		keepPreviousData: true,
```

Comment:
You might want to set `staleTime: Infinity` instead of `refetchOnWindowFocus` and `keepPreviousData`. That tells React Query that once the `HappychatAuth` data have been fetched, they never get stale and never need to be refetched. Seems that's exactly what we want here.